posts/2019/05/spyder-4-coming-closer-beta2-is-now-available.md

```diff
@@ -0,0 +1,148 @@
+Spyder development team has been working really hard on its next release,
+4.0 Beta2. The past couple of months we added the long awaited dark theme
+for the entire interface, we switched our entire code completion and linting
+architecture to the
+[Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
+which opens the door to support many other
+programming languages, and we merged several other goodies!
+
+## Dark Theme
+
+This has been a
+[long awaited feature](https://github.com/spyder-ide/spyder/issues/2350)
+and will be the default theme for this new version. Users can still select the
+traditional OS specific theme by going to the **Appearance** section on our
+**Preferences** dialog.
+
+This enhancement was enabled by the work of
+[Colin Duquesnoy](https://github.com/ColinDuquesnoy) and
+[Daniel Pizzeta](https://github.com/DPizzeta)
+and their [QDarkStyle](https://github.com/ColinDuquesnoy/QDarkStyleSheet/)
+package. The Spyder team is now actively collaborating with Colin and Daniel
+to pursue the release of QDarkStyle 3.x, which will be using the
+[qtsass](https://github.com/spyder-ide/qtsass/)
+package to harness the power of SASS/SCSS and allow users to fully customize
+the theme dynamically.
+
+A picture is worth a thousand words, so without further ado, here it is:
+
+![Spyder Dark style](/images/spyder-qdarkstyle.png)
+
+Pretty, right :-) ?
+
+<!-- TEASER_END -->
+
+## Language Server Protocol (LSP) implementation
+
+The LSP was created by Microsoft for Visual Studio Code to standardize the
+way in which development tools (i.e. editors and IDEs) communicate with
+servers that provide code completion, linting and related facilities for
+different programming languages. With this, as
+[they](https://microsoft.github.io/language-server-protocol/) describe it:
+
+>A single Language Server can be reused in multiple development
+tools, which in turn can support multiple languages with minimal effort.
+LSP is a win for both language providers and tooling vendors!
+
+Since our 4.0beta2 version, Spyder is now one of such development tools! We
+developed our own client to communicate with any server that implements the
+LSP v3.0 through a transport layer that uses ZeroMQ sockets. Code completion,
+signatures and docstring extraction were rewritten to take advantage of this
+architecture, and mouse hovers were added too (other LSP features, such as
+workspace functionality, will come in future betas).
+
+Our current support is geared towards Python, using the great
+[Python-Language-Server](https://github.com/palantir/python-language-server)
+package. This has allowed us to provide fine-grained configurability for
+[Pycodestyle](http://pycodestyle.pycqa.org/en/stable/) and
+[Pydocstyle](http://www.pydocstyle.org/en/stable/) options, and in future
+betas we’ll also add the ability to use and configure code formatters such
+as yapf and autopep8.
+
+![Hover hint and calltips](/images/spyder-hover-hint-calltip.png)
```

Comment:
@goanpeca, please add an screenshot below this one with the `Code style` and `Dostring style` tabs of the `Completion and linting` entry in our Preferences.